lib/simp/cli/environment/omni_env_controller.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+require 'simp/cli/environment/env'
+require 'simp/cli/environment/puppet_dir_env'
+require 'simp/cli/environment/secondary_dir_env'
+require 'simp/cli/environment/writable_dir_env'
+
+# Puppetfile helper namespace
+module Simp::Cli::Environment
+  # Controller class to manage SIMP Omni environments
+  class OmniEnvController
+    def initialize(opts = {}, env = nil)
+      @opts = opts
+      @environments = {}
+      @opts[:types].each do |type, data|
+        # TODO: honor backends?
+        # TODO: refactor into a Factory
+        base_env_path = data[:environmentpath] || fail(ArgumentError, 'ERROR: no :environmentpath in opts')
+        opts_data = data.reject { |k, _v| k == :enabled }
+        case type
+        when :puppet
+          @environments[:puppet]    = PuppetDirEnv.new(env, base_env_path, opts_data)
+        when :secondary
+          @environments[:secondary] = SecondaryDirEnv.new(env, base_env_path, opts_data)
+        when :writable
+          @environments[:writable]  = WritableDirEnv.new(env, base_env_path, opts_data)
+        else
+          fail("ERROR: Unrecognized environment type '#{env_type}'")
+        end
+      end
+    end
+
+    # Create a new environment for each environment type
+    def create
+      each_environment 'create' do |_env_type, env_obj|
+        env_obj.create
+      end
+
+      # ensure environments are correct after creating them
+      fix
+    end
+
+    # Update environment
+    def update
+      fail NotImplementedError
+    end
+
+    # Remove environment
+    def remove
+      fail NotImplementedError
+    end
+
+    # @return [Hash<Simp::Cli::Environment::Env>] current environments
+    def list
+      fail NotImplementedError
+    end
+
+    # Fix consistency of environment
+    def fix
+      each_environment 'fix' do |env_type, env_obj|
+        if @opts[:types][env_type].fetch(:strategy,'') == :link
+          warn("INFO: (action: fix) skipping fix of #{env_type} environment because strategy is :link")
```

Comment:
This to me is a debug message.  I don't need to do anything about it and really don't need to know about it unless I am curious about details.